policy/lib/rule_data.rego

```diff
@@ -0,0 +1,38 @@
+package lib
+
+#
+# Will define built-in default values here.
+# They can be overridden if required.
+#
+rule_data_defaults := {
+	#
+	# Used in release/attestation_type
+	"known_attestation_types": ["https://in-toto.io/Statement/v0.1"],
+	#
+	# Used in release/slsa_provenance_available
+	"allowed_predicate_types": ["https://slsa.dev/provenance/v0.2"],
+	#
+	# Used in release/slsa_build_build_service
+	"allowed_builder_ids": ["https://tekton.dev/chains/v2"],
+	#
+	# For the tests only. Not sure how else to get to 100% coverage.
+	"_test_key_3": 10,
+}
+
+#
+# Returns the "first found" of the following:
+#   data.rule_data_custom[key_name]
+#   data.rule_data[key_name]
+#   rule_data_defaults[key_name]
+#
+# And falls back to an empty list if the key is not found anywhere.
```

Comment:
Is an empty list the right value? I suppose if we do reach the default value, something is up. Is there a `panic` in rego? :thinking:

Reply:
All of the rule data we got are lists so a list is likely the right choice here.

Reply:
The argument for a null is that it will probably be more likely to throw an error if you typoed the key for example.